packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapController.m

```diff
@@ -506,7 +513,7 @@ - (void)setMyLocationButtonEnabled:(BOOL)enabled {
 }
 
 - (NSString *)setMapStyle:(NSString *)mapStyle {
-  if (mapStyle == (id)[NSNull null] || mapStyle.length == 0) {
+  if (mapStyle.length == 0) {
     self.mapView.mapStyle = nil;
```

Comment:
should we clear the error here?

Reply:
The caller is responsible for using the return value to update the error state.

Reply:
if the caller successfully set the style (to be nil), shouldn't we mark our internal state as no error? (The error ivar is used to track the error happened in previous setStyle call right?)

Reply:
I don't understand what code specifically you are referring to as "we" and "our" in that statement, so I don't know how to answer.

> (The error ivar is used to track the error happened in previous setStyle call right?)

The same code that set it previously would clear it. I'm not sure what the call sequence you are concerned about here is exactly.